sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/TestXmlData.scala

```diff
@@ -68,4 +68,444 @@ private[xml] trait TestXmlData {
+          |        </item>
+          |        <item>
+          |          <inner1>str4</inner1>
+          |        </item>
+          |      </field1>
+          |      <field2>
+          |        <array>
+          |          <item>5</item><item>6</item>
+          |        </array>
+          |        <array>
+          |          <item>7</item><item>8</item>
+          |        </array>
+          |      </field2>
+          |    </item>
+          |  </complexArrayOfStruct>
+          |  <arrayOfArray1>
+          |  <array>
+          |    <item>5</item>
+          |  </array>
+          |</arrayOfArray1>
+          |<arrayOfArray1>
+          |  <array>
+          |    <item>6</item><item>7</item>
+          |  </array>
+          |  <array>
+          |    <item>8</item>
+          |  </array>
+          |</arrayOfArray1>
+          |  <arrayOfArray2>
+          |  <array>
+          |    <item>
+          |      <inner1>str1</inner1>
+          |    </item>
+          |  </array>
+          |</arrayOfArray2>
+          |<arrayOfArray2>
+          |  <array/>
+          |  <array>
+          |    <item>
+          |      <inner2>str3</inner2>
+          |      <inner2>str33</inner2>
+          |    </item>
+          |    <item>
+          |      <inner2>str4</inner2>
+          |      <inner1>str11</inner1>
+          |    </item>
+          |  </array>
+          |</arrayOfArray2>
+          |<arrayOfArray2>
+          |  <array>
+          |    <item>
+          |      <inner3>
+          |        <inner4>2</inner4>
+          |        <inner4>3</inner4>
+          |      </inner3>
+          |      <inner3/>
+          |    </item>
+          |  </array>
+          |</arrayOfArray2>
+          |</ROW>
+          |""".stripMargin :: Nil
+
+  def nullsInArrays: Seq[String] =
+    """<ROW>
+          <field1>
+            <array>
+              <item/>
+            </array>
+            <array>
+              <array>
+                <array>
+                  <item>Test</item>
+                </array>
+              </array>
+            </array>
+          </field1>
+        </ROW>""" ::
+    """
+        <ROW>
+          <field2>
+            <item/>
+            <array>
+              <item>
+                <Test>1</Test>
+              </item>
+            </array>
+          </field2>
+        </ROW>""" ::
+    """
+        <ROW>
+          <field3>
+            <array>
+              <item/>
+            </array>
+            <array>
+              <item>
+                <Test>2</Test>
+              </item>
+            </array>
+          </field3>
+        </ROW>""" ::
+    """
+        <ROW>
+          <field4>
+            <array>
+              <item/>
+              <array>
+                <item>1</item>
+                <item>2</item>
+                <item>3</item>
+              </array>
+            </array>
+          </field4>
+        </ROW>""" :: Nil
+
+  def corruptRecords: Seq[String] =
```

Comment:
unused

Reply:
I plan to add tests in a follow-up PR. I will remove this for clarity in this PR.